### ISSUE TITLE
Depth map

### ISSUE THREAD

**Issue by kaiaa7** [lwwu2/diver#5]
Hi, thank you very much for your great work!
I can get the correct rgb image with your code, but the depth map is not very accurate. I am using the following code to render the depth map.

```
color, sigma, mask, ts = model(ray_o, ray_d)
depth = (ts*mask*weight).sum(1)
```

The rendered result(left) and ground truth(right) of the depth map is as follows.
<img width="400" alt="1" src="https://user-images.githubusercontent.com/110881688/183566476-273c81f9-84ee-4a2f-91a4-93b4f3675587.png">
The rendered result is more like a grayscale image but not depth map, Do you have any suggestions for this?

**Comment by lwwu2** [comment]
Hi, thank you for your interest in our work!

The 'depth' calculated in this way is actually the distance from camera center to the expected intersection point or the expected ray step. Sorry for the confusion. The conversion to the actual depth map should be something like $z=d/\sqrt{(\frac{u+c_x}{f})^2+(\frac{v+c_y}{f})^2+1}$, where $z,d$ are the actual depth and ray step; $(u,v),(c_x,c_y),f$ are screen space coordinate, optical center, and focal length.

Even without the conversion, the left image still does not look like a ray step map. Can you double check if it is the ray step map being displayed? Otherwise, the model is misinterpreting shading and geometry which is possible.

**Comment by kaiaa7** [comment]
Thanks for the response, I checked my code and I could not find anything that should be modified, so I ran your code again. Because I just wanted to see the effect of the depth map, I only ran the first step, `python train.py --experiment_name drum_coarse --device 0 --max_epochs 10`. One of the rendered results in tensorboard(left) and ground truth(right) is as follows.
<img width="430" alt="2" src="https://user-images.githubusercontent.com/110881688/184079798-e0182ad9-fd4d-4acc-a993-62b7b91b209d.png">
 I think the rendered result is similar to what I got before, and it also doesn't look like a ray step map. Do you have any suggestions for this?

**Comment by lwwu2** [comment]
The reason why it looks odd is because pytorch lightning did not normalize the input to [0,1] when displaying it. If you rescale the the map to [0,1], you should get the expected ray step map. Meanwhile, if I remember correctly, the ground truth from the synthetic nerf dataset actually clip the value instead of rescale the value to be in [0,1].

**Comment by kaiaa7** [comment]
Thanks for the response. After I scaled the depth map obtained by the train.py to [0,1], I think I got the expected ray step map. The result is as follows. 

<img width="200" alt="1-drum-depth-64-5" src="https://user-images.githubusercontent.com/110881688/185393232-e3a4529e-e9e8-4d22-af6c-fd1abc54e486.png">

I will continue to check the details of my code to solve the problem that the depth map obtained by my code is incorrect.
